baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/PreferGuavaVisibleForTesting.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+// modeled off of
+// https://github.com/google/error-prone/blob/master/core/src/main/java/com/google/errorprone/bugpatterns/checkreturnvalue/UsingJsr305CheckReturnValue.java
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.ImportTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.suppliers.Supplier;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ImportTree;
+import com.sun.tools.javac.code.Type;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Checker that recommends using Guava's version of {@code @VisibleForTesting} over other copies.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(summary = "Prefer Guava's @VisibleForTesting over other versions.", severity = SeverityLevel.WARNING)
+public final class PreferGuavaVisibleForTesting extends BugChecker implements ImportTreeMatcher {
+    private static final String GUAVA_VFT = "com.google.common.annotations.VisibleForTesting";
+    private static final Set<String> OTHER_VFTS = Set.of(
+            "org.apache.hadoop.thirdparty.com.google.common.annotations.VisibleForTesting",
+            "org.jetbrains.annotations.VisibleForTesting",
+            "com.facebook.react.common.annotations.VisibleForTesting",
+            "org.apache.arrow.util.VisibleForTesting",
+            "org.assertj.core.util.VisibleForTesting",
+            "org.apache.flink.annotation.VisibleForTesting",
+            "com.cronutils.utils.VisibleForTesting",
+            "com.google.api.client.repackaged.com.google.common.annotations.VisibleForTesting",
+            "org.sparkproject.guava.annotations.VisibleForTesting",
+            "org.apache.hadoop.classification.VisibleForTesting",
+            "org.jheaps.annotations.VisibleForTesting",
+            "io.debezium.annotation.VisibleForTesting",
+            "graphql.VisibleForTesting",
+            "avro.shaded.com.google.common.annotations.VisibleForTesting");
```

Comment:
I'd prefer to avoid explicitly listing these and just forbid any annotation named "VisibleForTesting" that isn't the error-prone one.

Reply:
Agree with Patrick, I think we may also want to consider blocking some of these repackaged imports in the baseline checkstyle as well by expanding `IllegalImports`:

https://github.com/palantir/gradle-baseline/blob/f797c67965ecce0d3714ad3a47884e87a898732d/.baseline/checkstyle/checkstyle.xml#L127-L132

Reply:
Removed the deny list of imports and instead now look for all imports with the same class name.

Is it a requirement to change Checkstyle at the same time as ErrorProne, or can they be independent?  The EP change seems better to me, because it can suggest and apply fixes.